src/HLS/binding/module/fu_binding.cpp

```diff
@@ -1187,7 +1191,56 @@ void fu_binding::add_to_SM(const HLS_managerRef HLSMgr, const hlsRef HLS, struct
          manage_module_ports(HLSMgr, HLS, SM, FU, 0);
          memory::propagate_memory_parameters(FU, HLS->datapath);
       }
+      if(HLSMgr->unused_interfaces.find(HLS->functionId) != HLSMgr->unused_interfaces.end())
+      {
+         const auto HLS_D = HLSMgr->get_HLS_device();
+         const auto TechMan = HLS_D->get_technology_manager();
+         const auto& res_pair_set = HLSMgr->unused_interfaces.at(HLS->functionId);
+         for(const auto& [UPlibrary, FUName] : res_pair_set)
+         {
+            auto fu_name = FUName;
+            technology_nodeRef fuObj = TechMan->get_fu(fu_name, UPlibrary);
+            const auto structManager_obj = GetPointer<functional_unit>(fuObj)->CM;
+            THROW_ASSERT(structManager_obj, "unexpected condition");
+            const auto has_to_be_generated = GetPointer<module>(structManager_obj->get_circ())
```

Comment:
Are you sure you would not like to write a unique method for this and call it two times instead of copy-pasting the same code? 😄

Reply:
The following commit will fix it.